listeners/syncprogress.go

```diff
@@ -0,0 +1,82 @@
+package listeners
+
+import (
+	"github.com/planetdecred/dcrlibwallet"
+	"github.com/planetdecred/godcr/wallet"
+)
+
+type SyncProgress struct {
+	SyncStatus chan wallet.SyncStatusUpdate
+}
+
+func NewSyncProgress(syncStatus chan wallet.SyncStatusUpdate) *SyncProgress {
+	return &SyncProgress{
+		SyncStatus: syncStatus,
+	}
+}
+
+func (sp *SyncProgress) OnSyncStarted(wasRestarted bool) {
+	sp.sendNotification(wallet.SyncStatusUpdate{
+		Stage: wallet.SyncStarted,
+	})
+}
+
+func (sp *SyncProgress) OnPeerConnectedOrDisconnected(numberOfConnectedPeers int32) {
+	sp.sendNotification(wallet.SyncStatusUpdate{
+		Stage:          wallet.PeersConnected,
+		ConnectedPeers: numberOfConnectedPeers,
+	})
+}
+
+func (sp *SyncProgress) OnCFiltersFetchProgress(cfiltersFetchProgress *dcrlibwallet.CFiltersFetchProgressReport) {
+	sp.sendNotification(wallet.SyncStatusUpdate{
+		Stage:          wallet.CfiltersFetchProgress,
+		ProgressReport: cfiltersFetchProgress,
+	})
+}
+
+func (sp *SyncProgress) OnHeadersFetchProgress(headersFetchProgress *dcrlibwallet.HeadersFetchProgressReport) {
+	sp.sendNotification(wallet.SyncStatusUpdate{
+		Stage: wallet.HeadersFetchProgress,
+		ProgressReport: wallet.SyncHeadersFetchProgress{
+			Progress: headersFetchProgress,
+		},
```

Comment:
Please let's improve this while we're here; the following just doesn't make sense:
```
type SyncHeadersFetchProgress struct {
	Progress *dcrlibwallet.HeadersFetchProgressReport
}
```

Either do `type SyncHeadersFetchProgress *dcrlibwallet.HeadersFetchProgressReport` or just use `*dcrlibwallet.HeadersFetchProgressReport` directly.

Reply:
Applies to a few other such types in the wallet/listener.go file.